app/presenters/workflows_presenter.rb

```diff
@@ -13,6 +13,17 @@ def workflows_props
     }
   end
 
+  def workflow_options_by_purchase_props(purchase:)
+    seller.workflows.alive.published
+      .joins(:installments)
+      .merge(Installment.alive.published)
+      .includes(:base_variant)
+      .distinct
+      .select { |workflow| workflow.applies_to_purchase?(purchase) }
+      .sort_by(&:name)
+      .map { |workflow| WorkflowPresenter.new(seller:, workflow:).workflow_option_props }
```

Comment:
1. `.joins(:installments).merge(Installment.alive.published)` filters workflows that have at least one alive, published installment.
2. `.includes(:base_variant)` eager loads `base_variant` to avoid N+1 queries. `applies_to_purchase?` accesses `workflow.base_variant` for variant-type workflows so without eager loading, each workflow would trigger a separate query.
3. `.distinct` prevents duplicate workflows when a workflow has multiple matching installments from the join

TL;DR: The joins/merge filter at the DB level, `.includes(:base_variant)` prevents N+1 when `applies_to_purchase?` accesses the association, and `.distinct` removes duplicates from the join.